lib/Core/Executor.cpp

```diff
@@ -896,120 +896,395 @@ Executor::fork(ExecutionState &current, ref<Expr> condition, bool isInternal) {
   // the value it has been fixed at, we should take this as a nice
   // hint to just use the single constraint instead of all the binary
   // search ones. If that makes sense.
-  if (res==Solver::True) {
+  if (NoCompression) {
+    if (res == Solver::True) {
+      TimerStatIncrementer timer(stats::forkTime);
```

Comment:
@rasoolmaghareh There is a lot of code that is being repeated between the bodies of the then, else if, and else clauses here. I suggest moving a large part of the repeated code outside of the if branching structure and encode only the parts that are different within the bodies of the if conditional.